includes/class-amp-autoloader.php

```diff
@@ -96,6 +97,7 @@ class AMP_Autoloader {
 		'AMP_Content'                        => 'includes/templates/class-amp-content',
 		'AMP_Content_Sanitizer'              => 'includes/templates/class-amp-content-sanitizer',
 		'AMP_Post_Template'                  => 'includes/templates/class-amp-post-template',
+		'AMP_DOM_Document'                   => 'includes/utils/class-amp-dom-document',
```

Comment:
Should this go into the `Amp\AmpWP` namespace now that we have it? Or is this premature?

Reply:
No, I think it makes sense, if we can agree that we'll use this new abstraction everywhere going forward.

I assume it should be `Amp\AmpWP\DOMDocument`, due to its importance?

Reply:
Sure, or what about `Amp\AmpWP\DOM\Document`?

Reply:
If we use a subnamespace like that, we would also add `DOMElementList` in there as well.

It would allow us to important the namespace only and then use the relative `DOM` subnamespace in the code, if we want:

```php
use Amp\AmpWP\DOM;

$dom = new DOM\Document();
$list = ( new DOM\ElementList() )
    ->add( $image1 )
    ->add( $image2 );
```

However, we might want to go with `Amp\AmpWP\Dom\Document` to match `Amp`. It would make the result less "shouty".